src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -91,11 +92,19 @@ public void CloneTo(X509Certificate2Collection collection)
                 return;
             }
 
+            var loadedCerts = new HashSet<X509Certificate2>();
+
             foreach (string filePath in Directory.EnumerateFiles(_storePath, PfxWildcard))
             {
                 try
                 {
-                    collection.Add(new X509Certificate2(filePath));
+                    var cert = new X509Certificate2(filePath);
+
+                    // If we haven't already loaded a cert .Equal to this one, copy it to the collection.
+                    if (loadedCerts.Add(cert))
```

Comment:
The hashcode value uses things which *could* not otherwise be loaded, but they're pretty common to load.  So it's frontloading a bit of work that would otherwise be a lazy-load.  (Though by "work" I mainly mean "marshal", since the real work was already done by OpenSSL when reading the bytes in the first place)